agatha/construct/semrep_util.py

```diff
@@ -302,49 +302,77 @@ def _get_flags(self, input_path:Path, output_path:Path)->List[str]:
     if self.word_sense_disambiguation:
       res.append("--word_sense_disambiguation")
     res.append("--xml_output_format")
-    res.append(str(input_path))
-    res.append(str(output_path))
+    if input_path is not None:
+      res.append(str(input_path))
+    if output_path is not None:
+      res.append(str(output_path))
```

Comment:
Lets make the parameters to `_set_flags` `:Optional[Path]=None` now that they can be `None`. This way its a little clearer from the signature what this function is expecting.
We might also want an assert that either both are set or both are none. Right now if `output_path` is set, but `input_path` is not, the process will try to read in from the output file.
A docstring would be helpful as well.